src/Fantomas.Core/CodePrinter.fs

```diff
@@ -3196,8 +3197,11 @@ let genTypeDefn (td: TypeDefn) =
                 +> optSingle (fun _ -> unindent) node.Accessibility
                 +> genMembers
 
-            let stroustrupWithoutMembers =
-                genAccessOpt node.Accessibility +> genRecordFields +> genMembers
+            let stroustrup =
+                genAccessOpt node.Accessibility
+                +> genRecordFields
+                +> onlyIf hasMembers (!- " with" +> indent)
```

Comment:
Re-use the `with` keyword when it is present.
It might contain trivia:

```fsharp
            let stroustrup =
                let withKw =
                    match typeName.WithKeyword with
                    | None ->  !- "with"
                    | Some withNode -> genSingleTextNode withNode
                
                genAccessOpt node.Accessibility
                +> genRecordFields
                +> onlyIf hasMembers (sepSpace +> withKw  +> indent)
                +> genMembers
```